<launch>

  <!-- Aruco param -->
  <arg name="markerSize"      default="0.095"/>    <!--(9.5cm) -->
  <arg name="ref_frame"       default=""/>  <!-- leave empty and the pose will be published wrt param parent_name -->

  <!-- overhead camera -->
  <include file="$(find usb_cam)/launch/usb_cam-test.launch" />

  <!-- image rectification -->
  <group ns="usb_cam">
    <node pkg="image_proc" type="image_proc" name="image_proc"/>
  </group>

  <!-- detect aruco -->
  <node pkg="aruco_ros" type="marker_publisher" name="aruco_marker_publisher" output ="screen">
    <param name="image_is_rectified" value="True"/>
    <param name="marker_size"        value="$(arg markerSize)"/>
    <param name="reference_frame"    value="$(arg ref_frame)"/> 
    <param name="use_camera_info"    value="true"/> 
    <remap from="/camera_info" to="/usb_cam/camera_info" />
    <remap from="/image" to="/usb_cam/image_rect" />
  </node>

  <!-- image_view -->
  <node ns = "aruco_display" name="image_view" type="image_view" pkg="image_view" output="screen">
    <remap from="image" to="/aruco_marker_publisher/result"/>
  </node>

</launch>